### DISCUSSION TITLE
Cannot connect via PnP Powershell to give access to SP to Endra ID APP

### DISCUSSION THREAD

**Discussion by OrginalError24Nova** [pnp/powershell#4572] [General]
Can someone help on this,
We have Entra ID APP, which has SP access. Below is the script we have used to give the access to SP.
Install-Module PnP.PowerShell
Import-Module PnP.PowerShell

$siteUrl = ""
$clientId = "" #clientid of app you just registered
$certThumbprint = "" #App with Sites.Selected permission
$tenant = ""

Connect-PnPOnline -Url $siteUrl -Interactive

$writeperm = Grant-PnPAzureADAppSitePermission -Permissions Write -Site $siteUrl -AppId $clientId -DisplayName """
$PermissionId = Get-PnPAzureADAppSitePermission -AppIdentity $clientId

Set-PnPAzureADAppSitePermission -Site 

(($PermissionId).Id) -Permissions FullControl

Now we need to give access to new SP site for this already created Endra ID APP.
But getting error that PNP connection not suppoerted.

**Comment by waaromikniet** [comment]
When you use "Connect-PnPOnline -Url $siteUrl -Interactive" you make use of the deleted PnP.PowerShell entra app. You should pass your own app clientid and make the connection.

Could you comment the exact commands and error?
